### PR TITLE
Add command to navigate to site editor

### PR DESCRIPTION
**Pull request by benazeer-ben**
## What?

Enhancement for the scenario mentioned in https://github.com/WordPress/gutenberg/issues/61460

## Why?

Currently there is no command option to go back to site editor from edit post or page views.

## How?

Added command to navigate to Site Editor.

## Testing Instructions

- Open the WordPress Editor
- Go to Posts or Pages in the WordPress admin.
- Open a new or existing post/page to enter the block editor.
- Open the Command Palette
- Access the Command Palette by pressing Cmd + K on Mac or Ctrl + K on Windows, or via the menu if available.
- In the Command Palette, look for a command labeled something like "Go to Site Editor".
- Ensure that this command is listed among the available commands.
- Execute the Command
- Select the Go to Site Editor command from the Command Palette.
- Confirm that executing the command redirects you to the Site Editor interface.

## Screenshots or screencast 

https://github.com/user-attachments/assets/e39219d2-195f-4eda-80ce-2ddc73eb6c14

### CODE REVIEW

**Comment by annezazu** [comment]
This works!

https://github.com/user-attachments/assets/8acd8e96-0d4b-4a5d-a9d2-079dcaadccf4

With that said, I wonder about the naming. Can we go with "Open the Site Editor" as that aligns with some of the "open" language outlined here https://github.com/WordPress/gutenberg/issues/50407 ? I also am curious about the icon used and wonder if it should be something else:

Screenshot 2024-11-05 at 9 59 38 AM

@WordPress/gutenberg-design for any quick thoughts on the icon!

**Comment by richtabor** [comment]
The command should also not be available when you're already in the Site Editor: 

CleanShot 2024-11-05 at 14 11 56

> @WordPress/gutenberg-design for any quick thoughts on the icon!

Let's just not use an icon, all commands don't need one.

File: packages/editor/src/components/commands/index.js [hunk 1]
```diff
@@ -253,7 +253,18 @@ function useEditorCommandLoader() {
 			},
 		} );
 	}
-
+	const isInSiteEditor =
+		window.location.pathname.includes( 'site-editor.php' );
```
Comment (t-hamano) [inline packages/editor/src/components/commands/index.js]:
```suggestion
	const isSiteEditor = getPath( window.location.href )?.includes(
		'site-editor.php'
	);
```

Please import `getPath` from the `@wordpress/url` package.

File: packages/editor/src/components/commands/index.js [hunk 2]
```diff
@@ -253,7 +253,18 @@ function useEditorCommandLoader() {
...
+	if ( ! isInSiteEditor ) {
+		commands.push( {
+			name: 'core/go-to-site-editor',
+			label: __( 'Open Site Editor' ),
+			callback: ( { close } ) => {
+				close();
+				window.location.href = '/wp-admin/site-editor.php';
```
Comment (t-hamano) [inline packages/editor/src/components/commands/index.js]:
```suggestion
				document.location = 'site-editor.php';
```

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

One thing to note here is that classic themes don't have access to the site editor root, so we can't add this command in classic themes.

Here's an example of code to detect if it's a block theme:

https://github.com/WordPress/gutenberg/blob/7ff76796b8fd7027a3f53933e1f9e5f2917ea2c8/packages/edit-site/src/components/add-new-pattern/index.js#L51-L53

We could add a similar code to [the existing useSelect hook](https://github.com/WordPress/gutenberg/blob/7ff76796b8fd7027a3f53933e1f9e5f2917ea2c8/packages/editor/src/components/commands/index.js#L34-L65).

**Comment by jameskoster** [comment]
Do we refer to the "Site Editor" anywhere else in the UI? I recognise it's a familiar term to wp community members, but otherwise we're introducing _another_ name for the same thing. The wp-admin navigation refers to this section simply as "Editor" (under "Appearance"). The root panel is titled "Design". 

> The command should also not be available when you're already in the Site Editor:

I think the purpose is to go from the full screen editor (in either post or site editor) to the site editor _root_? This also suggests to me that the naming isn't quite right... "Site editor" means different things to different people.

**Comment by jasmussen** [comment]
I'm still slightly partial to Site view and Edit view, though I recognize they haven't really caught on:

Previous version

Your point is valid enough. Another way to frame it, however, is: do we support aliases? If someone searches for "site editor" as a command, even if we chose not to surface the specific name in the palette, it would still be useful for fuzzy searching. We support aliases for blocks already, which is why all of these results show up in a search for "image":

Screenshot 2024-11-06 at 12 47 36

**Comment by t-hamano** [comment]
The term "Site Editor" is used in the following places, but I'm not sure if this term is official.

### Site Editor Mobile Layout > Site Hub Tooltip Label

mobile-navigation

### Site Editor > Page Screen > Welcome Guide

page-welcome-guide

### Site Editor > Welcome Guide Area Label

aria-label-site-editor

**Comment by t-hamano** [comment]
@benazeer-ben Thanks for the update!

As mentioned in [the comment](https://github.com/WordPress/gutenberg/pull/66722#pullrequestreview-2417378270), this command cannot be exposed to classic themes. Could you update this PR?


One more thing I noticed is that the site editor is not accessible to all users. As implemented [here](https://github.com/WordPress/gutenberg/blob/865a05834d3af46a6eecd142fe6acda40642756d/packages/core-commands/src/site-editor-navigation-commands.js#L300), the command should only be exposed if the user can create templates and it is a block theme.

**Comment by benazeer-ben** [comment]
Thanks for that suggestion @t-hamano 

Now the PR is updated with both the conditions.

**Review by t-hamano — APPROVED** [review]
From a code perspective, I think it's ready to ship, but should the phrase "Open Site Editor" be allowed? I don't think the Command Palette currently lets us set alias labels.

**Comment by ramonjd** [comment]
I'll merge this since @t-hamano has approved. We could update the copy later?

**Pull request merged**